vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -38,8 +38,8 @@ struct ccb
   std::vector<uint32_t> origin_index;
   CB::cb_class cb_label, default_cb_label;
   std::vector<bool> exclude_list, include_list;
-  std::vector<std::string> generated_interactions;
-  std::vector<std::string>* original_interactions;
+  std::vector<std::vector<uint8_t>> generated_interactions;
```

Comment:
Most of the usages of `uint8_t` should really be `namespace_index` as that is what they logically are. Also, `namespace_index` should be `uint8_t`

Reply:
Apologies, I didn't understand. Are you asking me to change uint8_t to namespace_index? Or the way round?

Reply:
`namespace_index` is just an alias, it should map to `uint8_t`, but the `namespace_index` should be used where appropriate

Reply:
Updated